src/Rule/Nested.php

```diff
@@ -167,10 +160,14 @@ private function prepareRules(iterable|object|string|null $source): void
 
         if ($source instanceof RulesProviderInterface) {
             $rules = $source->getRules();
-        } elseif (!$source instanceof Traversable && !is_array($source)) {
+        } elseif (is_string($source) && class_exists($source)) {
             $rules = (new AttributesRulesProvider($source, $this->rulesPropertyVisibility))->getRules();
-        } else {
+        } elseif (is_iterable($source)) {
             $rules = $source;
+        } else {
+            throw new InvalidArgumentException(
+                'Invalid rules object.'
```

Comment:
It's not necessarily an object. Can be a simple string or a string with not existing class name.

Reply:
Also it's a bit broad - for a user it can be hard to understand what exactly is wrong. We can have more specific exceptions with more clear messages, for example a separate exception for string type - that only existing class names are expected. But it can be improved later, with a friendly exception maybe.

Reply:
> Please, suggest message. I'm replace.

https://github.com/yiisoft/validator/pull/403/files#r1034697378